### PR TITLE
Implemnt missing features from pg_regress test suite

### PR DESCRIPTION
**Pull request by rdunklau**
I tried to implement each change in a self contained commit, let me know if things should be split more.

### CODE REVIEW

File: pglast/enums/__init__.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #
 
 # include/catalog
+from .pg_am import *     # noqa
```
Comment (lelit) [inline pglast/enums/__init__.py:10]: Remember to add the new `pg_am.rst` document in `docs/enums.rst`

**Comment by lelit** [comment]
Great, thank you! Just tell me when you are comfortable at merging this. Right now I'm quite busy on other things, so I won't work on pglast for another couple of days at least.

File: pglast/enums/__init__.py (hunk 1 above)
Comment (rdunklau) [reply pglast/enums/__init__.py:10]: Done ! Thank you.

**Comment by rdunklau** [comment]
I'll keep on adding cases, and let you know when I'm done.

**Comment by lelit** [comment]
Hi! I'm going to accept and merge #77 soon: it builds on top of your commits implementing missing printer functions.

**Comment by lelit** [comment]
Thank you Ronan, this has landed in the v3 branch.

**Pull request merged**